Upgrade docker-py to 7.1.0

saiarcot895 opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it

This version of docker-py has the fix for the issue caused by requests 2.32, and has the added benefit of upgrading to the latest version of docker-py.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

#### How to verify it

<!--
If PR needs to be backported, then the PR must be tested against the base branch and the earliest backport release branch and provide tested image version on these two branches. For example, if the PR is requested for master, 202211 and 202012, then the requester needs to provide test results on master and 202012.
-->

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

<!--
- Please provide tested image version
- e.g.
- [x] 20201231.100
-->

- [ ] <!-- image version 1 -->
- [ ] <!-- image version 2 -->

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

<!--
 Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

kperumalbfn commented on build_debian.sh:
```diff
@@ -537,7 +537,7 @@ sudo cp files/image_config/pip/pip.conf $FILESYSTEM_ROOT/etc/pip.conf
 
 # docker Python API package is needed by Ansible docker module as well as some SONiC applications
-sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT pip3 install 'docker==6.1.1'
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT pip3 install 'docker==7.1.0'
```
Is this change required in other branches?

saiarcot895 replied:
I believe it is needed on 202311 and 202305 as well (at minimum). However, looking in the build pipelines, I see that the build for both is passing (without any version pinning). If it is actually needed, I can add the labels here.

lguohan commented:
/azpw ms_conflict

anamehra commented:
Hi @saiarcot895 , any update on double commit? I see requests label added last week but no commit yet. This is affecting builds on 202311. Thanks

saiarcot895 commented:
@anamehra I had added but then removed the requests labels, because the build pipelines for 202305 and 202311 were passing. However, I now see that in the pipelines I was looking at, Python 2 and Python 3 package versions are pinned for non-master branches, which means it would not run into this error.

I've added the request labels for 202305 and 202311.

krismarvell commented:
HI @saiarcot895 I am running into this error for innovium build in 202211 branch. Can we backport this into 202211 as well?

mssonicbld commented:
Cherry-pick PR to 202211: https://github.com/sonic-net/sonic-buildimage/pull/20359

mssonicbld commented:
Cherry-pick PR to 202305: https://github.com/sonic-net/sonic-buildimage/pull/20360

liushilongbuaa commented:
> Cherry-pick PR to 202211: #20359

Builb failed.
```
Writing /usr/local/lib/python2.7/dist-packages/scapy-2.2.0_dev.egg-info
Cloning into 'sflowtool'...
configure.ac:4: error: Autoconf version 2.71 or higher is required
configure.ac:4: the top level
autom4te: /usr/bin/m4 failed with exit status: 63
aclocal: error: echo failed with exit status: 63
autoreconf: aclocal failed with exit status: 63
The command '/bin/sh -c rm -rf /debs     && apt-get -y autoclean     && apt-get -y autoremove     && rm -rf /var/lib/apt/lists/*     && wget --https-only https://bootstrap.pypa.io/pip/2.7/get-pip.py     && python get-pip.py     && rm -f get-pip.py     && pip install setuptools     && pip install supervisor     && pip install ipython==5.4.1     && git clone https://github.com/p4lang/scapy-vxlan.git     && cd scapy-vxlan     && python setup.py install     && cd ..     && rm -fr scapy-vxlan     && git clone https://github.com/sflow/sflowtool     && cd sflowtool     && ./boot.sh     && ./configure     && make     && make install     && cd  ..     && rm -fr sflowtool     && git clone https://github.com/dyninc/OpenBFDD.git     && cd OpenBFDD     && ./autogen.sh     && ./configure     && make     && make install     && cd  ..     && rm -fr OpenBFDD     && wget https://github.com/nanomsg/nanomsg/archive/1.0.0.tar.gz     && tar xvfz 1.0.0.tar.gz     && cd nanomsg-1.0.0        && mkdir -p build          && cd build                && cmake ..                && make install            && ldconfig                && cd ../..                && rm -fr nanomsg-1.0.0     && rm -f 1.0.0.tar.gz      && pip install cffi        && pip install nnpy        && pip install dpkt        && pip install ipaddress     && pip install pysubnettree     && pip install paramiko     && pip install flask       && pip install exabgp==3.4.17    && pip install pyaml       && pip install pybrctl pyro4 rpyc yabgp     && pip install unittest-xml-reporting     && pip install pyrasite     && pip install retrying     && mkdir -p /opt           && cd /opt                 && wget https://raw.githubusercontent.com/p4lang/ptf/master/ptf_nn/ptf_nn_agent.py' returned a non-zero code: 63
[  FAIL LOG END  ] [ target/docker-ptf.gz ]

```

mssonicbld commented:
Cherry-pick PR to 202311: https://github.com/sonic-net/sonic-buildimage/pull/21092
